geode-core/src/main/java/org/apache/geode/internal/tcp/ConnectionTable.java

```diff
@@ -443,32 +434,14 @@ Connection getThreadOwnedConnection(InternalDistributedMember id, long startTime
     if (m == null) {
       // First time for this thread. Create thread local
       m = new HashMap();
-      synchronized (threadConnMaps) {
-        if (closed) {
-          owner.getCancelCriterion().checkCancelInProgress(null);
-          throw new DistributedSystemDisconnectedException("Connection table is closed");
-        }
-        // check for stale references and remove them.
-        for (Iterator it = threadConnMaps.iterator(); it.hasNext();) {
-          Reference r = (Reference) it.next();
-          if (r.get() == null) {
-            it.remove();
-          }
-        }
-        threadConnMaps.add(new WeakReference(m));
-      }
       threadOrderedConnMap.set(m);
```

Comment:
Now with all the other stuff gone we could replace this with:
```
static final ThreadLocal<Map<DistributedMember, InternalConnection>> threadOrderedConnMap =
      ThreadLocal.withInitial(HashMap::new);
```

Reply:
I liked this idea initially but then I found a couple of existing places in the code that we call get on this thread local to see if the current thread has a map and if it does to do something. If we make this change then these calls (they are cleanup methods) can end up creating a new HashMap for threads that didn't have one. So for this PR and like to hold off on making this change.